.github/workflows/ci.yml

```diff
@@ -40,4 +40,46 @@ jobs:
       - name: verify go.mod and go.sum are consistent
         run : go mod tidy
       - name: Ensure nothing changed
-        run: git diff --exit-code
\ No newline at end of file
+        run: git diff --exit-code
+  tests:
+    name: Integration Tests
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Restore build cache
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache/go-build
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.mod') }}
+      - name: Set local Go version
+        run: |
+          VERSION=`cat .go-version | awk '{printf$1}'`
+          echo "go_version=$VERSION" >> $GITHUB_ENV
+      - name: Setup Go environment
+        uses: actions/setup-go@v2
+        with:
+          go-version: "${{ env.go_version }}"
```

Comment:
Non-blocker -- setup-go can actually
* Automatically cache Go modules between runs, no `actions/cache` step required.
* Automatically detect the [major.minor] version of Go from the go.mod file.

See the [internal] https://github.com/hashicorp/citool-example-project/blob/main/.github/workflows/test.yml#L11-L15 for an example of this in use.